### PR TITLE
Add logging level options.

### PR DESCRIPTION
**Pull request by menanteau**
Different logging levels can be set from the command using the --log option and from the input yaml configuration file

### CODE REVIEW

File: bin/DMHS_ATS_configurable [hunk 1]
```diff
@@ -45,13 +39,17 @@ def cmdline():
                         help="Port number for webserver")
     parser.add_argument("--send_efd_message", action="store_true", default=False,
                         help="Send and EDF message as well?")
+    parser.add_argument("--log", action="store", default='INFO',
+                         choices=["DEBUG","INFO","WARNING","ERROR","CRITICAL"],
```
Comment (timj) [inline bin/DMHS_ATS_configurable]: There is a flake8 violation here since you need a space after the comma.  Please set up continuous integration for PRs so that flake8 is automatically run.  I can help you with this.

File: bin/DMHS_ATS_configurable [hunk 2]
```diff
@@ -45,13 +39,17 @@ def cmdline():
                         help="Port number for webserver")
     parser.add_argument("--send_efd_message", action="store_true", default=False,
                         help="Send and EDF message as well?")
+    parser.add_argument("--log", action="store", default='INFO',
```
Comment (timj) [inline bin/DMHS_ATS_configurable]: I think I would prefer `--loglevel` or `--log-level` as the name of this parameter. That's what the parameter becomes internally.

File: bin/DMHS_ATS_configurable (hunk 1 above)
Comment (menanteau) [reply bin/DMHS_ATS_configurable]: Thanks for catching that. Yes, I'd like to setup continuous integration.

File: bin/DMHS_ATS_configurable [hunk 3]
```diff
@@ -45,13 +39,17 @@ def cmdline():
...
+                         choices=["DEBUG","INFO","WARNING","ERROR","CRITICAL"],
+                         help="Logging Level [DEBUG/INFO/WARNING/ERROR/CRITICAL]")
+
     # args = parser.parse_args()
     # Set the defaults of argparse using the values in the yaml config file
     parser.set_defaults(**conf_defaults)
     args = parser.parse_args(args=remaining_argv)
+    args.loglevel = getattr(logging, args.log.upper())
```
Comment (timj) [inline bin/DMHS_ATS_configurable]: The upper casing here is not doing anything because the choices above mandates that only upper case options are allowed. I think you need to add `type=str.upper` to your `add_argument()` call above so that lower case `debug` is converted to `DEBUG` before the choices are checked.

File: etc/conf/atTelemetry.yaml [hunk 4]
```diff
@@ -1,5 +1,8 @@
 # Configuration file for the AuxTelescope
 
+# logging level
```
Comment (timj) [inline etc/conf/atTelemetry.yaml]: Can you add `Default` here so that it reads `Default logging level`?

File: bin/DMHS_ATS_configurable (hunk 2 above)
Comment (menanteau) [reply bin/DMHS_ATS_configurable]: `--loglevel` was my original parameter, but the changed it to `--log `because it looked simpler. Happy to change it `--loglevel`

File: bin/DMHS_ATS_configurable [hunk 5]
```diff
@@ -45,13 +39,17 @@ def cmdline():
...
     return args
 
-
```
Comment (timj) [inline bin/DMHS_ATS_configurable]: Please put whitespace changes in a separate commit.

File: bin/DMHS_ATS_configurable (hunk 1 above)
Comment (timj) [reply bin/DMHS_ATS_configurable]: Sorry. I was doing a review but got distracted by a side conversation and ended up pushing "Add Single comment" all the time.

File: python/HeaderService/hutils.py [hunk 6]
```diff
@@ -358,7 +358,7 @@ def update_records(self, newdict, extname):
                 self.update_record(keyword, value, extname)
                 LOGGER.debug("Updating {}".format(keyword))
             except Exception:
-                LOGGER.debug("WARNING: Could not update {}".format(keyword))
+                LOGGER.warning("WARNING: Could not update {}".format(keyword))
```
Comment (timj) [inline python/HeaderService/hutils.py]: I would prefer that the changes of warning levels are made in a separate commit distinct from the commit that adds the ability to override the log level.

File: bin/DMHS_ATS_configurable (hunk 2 above)
Comment (timj) [reply bin/DMHS_ATS_configurable]: loglevel is consistent with what science pipelines are doing (pytest uses `log-level` so there isn't consistency for python).

File: etc/conf/atTelemetry.yaml (hunk 4 above)
Comment (menanteau) [reply etc/conf/atTelemetry.yaml]: Yes. Done.

File: python/HeaderService/hutils.py (hunk 6 above)
Comment (menanteau) [reply python/HeaderService/hutils.py]: Reverted to debug.

Comment (timj) [reply python/HeaderService/hutils.py]: It looks like `logging` suggests `.exception()` be used for logging exception information.

**Comment by timj** [comment]
@menanteau I had hoped you would be able to split the commit into separate commits: whitespace, flake8, addition of -loglevel, but I see the pull request has been merged so it is too late to fix now. `git add -p` is a very useful command that can be used to add subsets of changes to the index before committing. This lets you make unrelated changes but commit them as distinct commits. Slack me if you want more information on how to do this.

**Comment by menanteau** [comment]
@timj I'm sorry, but I misinterpreted your suggestion of putting the whitespace, etc in a different commit as a not belonging to the current branch. Reading it now it makes total sense as a different commit within the *same* branch. I get it now. But I guess is too late to fix it. Let me know if that is not the case.
